src/Uno.UI/UI/Xaml/Shapes/Ellipse.Android.cs

```diff
@@ -1,33 +1,28 @@
-﻿using Android.Graphics;
+﻿using Windows.Foundation;
+using Android.Graphics;
 using Uno.UI;
-using System;
-using System.Drawing;
-using Size = Windows.Foundation.Size;
 
 namespace Windows.UI.Xaml.Shapes
 {
 	public partial class Ellipse : ArbitraryShapeBase
 	{
-		public Ellipse()
+		protected override Size MeasureOverride(Size availableSize)
 		{
-			//Set default stretch value
-			this.Stretch = Windows.UI.Xaml.Media.Stretch.Fill;
+			base.MeasureOverride(availableSize);
+
+			// Ellipse will only ask for its "minimum" defined size.
+			return this.GetMinMax().min.AtLeastZero();
 		}
 
-		protected override Android.Graphics.Path GetPath()
+		protected override Android.Graphics.Path GetPath(Size availableSize)
 		{
-			var minMaxInLogicalPixels = this.GetMinMax();
-
-			var viewGroup = this as UnoViewGroup;
-			var finalSizeInPhysicalPixels =
-				new Size(viewGroup.Width, viewGroup.Height)
-					.AtMost(minMaxInLogicalPixels.max.LogicalToPhysicalPixels())
-					.AtLeast(minMaxInLogicalPixels.min.LogicalToPhysicalPixels());
-
-			var bounds = new RectF(0, 0, (float)finalSizeInPhysicalPixels.Width, (float)finalSizeInPhysicalPixels.Height);
+			var bounds = availableSize.LogicalToPhysicalPixels();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'availableSize' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4946529)